Replaces namespace prefix w/global namespace in payload for CreateTopicBodyImpl (fixes #34880)

haraldk opened the pull request:
# Description

This is a fix for #34880: Updating TopicProperties w/AuthorizationRules returns 400 Bad request.

The problem is that the Azure service does not support AuthorizationRules with XML namespace prefix, like the Java SDK creates by default (even if the XML itself is fine). This fix uses a common "hack" already in `com.azure.messaging.servicebus.administration.implementation.ServiceBusManagementSerializer` and used for fixing this for other types of payloads, that strips the namespaces, and instead sets a new default XML namespace. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

haraldk commented:
I’m still awaiting some form of human response from the Azure team on this. 

After more than two months, I no longer have much expectations, but it would still be nice to have this issue fixed.

anuchandy commented:
@haraldk I'm sorry that we missed to follow-up your contribution. The changes look good to me, would you be able to resolve the conflict, so that we can take this improvement.

haraldk commented:
@anuchandy Thanks. 👍🏻 

I've resolved the conflict and updated the PR now (rebased on current main and fixed the test after the changes). 

Would be nice to have this PR merged, even if we have had to make workarounds for this issue long ago...
